src/modules/hooks/useFetchAppointments.ts

```diff
@@ -0,0 +1,31 @@
+import { useAppDispatch, useAppSelector } from 'hooks';
+import {
+  appointmentsForDoctorSlice, appointmentsForPatientSlice, selectProfile,
+} from 'redux/reducers';
+import { useEffect } from 'react';
+import { appointmentsPerPage } from 'modules/hooks/constants';
+import { ROLES } from 'routes/constants';
+
+export const useFetchAppointments = (page: number, searchTerm?: string) => {
+  const dispatch = useAppDispatch();
+  const { id: userId, roleName } = useAppSelector(selectProfile);
+  const slice = roleName === ROLES.DOCTOR ? appointmentsForDoctorSlice : appointmentsForPatientSlice;
+
+  useEffect(() => {
```

Comment:
Looks like this effect will be called every time when search value will be updated.
Please, check debounce in hook usage to avoid this behaviour https://usehooks.com/useDebounce/

Reply:
The delay effect is written in saga and it has the same behavior as useDebounce, doesn't it?